test/framework/control_plane.go

```diff
@@ -34,84 +34,191 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const (
-	// eventuallyInterval is the polling interval used by gomega's Eventually
-	eventuallyInterval = 10 * time.Second
+// Package globals used to populate default values.
+// Can be updated by consumer to change default values.
+var (
+	// DefaultTimeout is used for the default timeout for every `Eventually` call.
+	DefaultTimeout = 10 * time.Minute
+
+	// DefaultPollInterval is used as the default for how frequently `Eventually` polls.
+	DefaultPollInterval = 10 * time.Second
 )
 
-// ControlplaneClusterInput defines the necessary dependencies to run a multi-node control plane cluster.
-type ControlplaneClusterInput struct {
-	Management        ManagementCluster
-	Cluster           *clusterv1.Cluster
-	InfraCluster      runtime.Object
-	Nodes             []Node
-	MachineDeployment MachineDeployment
-	RelatedResources  []runtime.Object
-	CreateTimeout     time.Duration
-	DeleteTimeout     time.Duration
+// Interfaces to scope down client.Client
```

Comment:
Is there a specific use case you have in mind for this?

I'm assuming it's to attempt to make it easier to create a fake client for use with the framework, but since it's an e2e framework, I would expect it to be operating against a real client anyway.

Reply:
Partially that *if* the situation arises that we need some unit tests for some complex logic that's it's easier to write and partially that it's easier to understand exactly what the test needs in order to run. If this uses a `client.Client` then every test could be making use of the entire interface, which is somewhat misleading.